test/e2e/flowaggregator_test.go

```diff
@@ -500,6 +491,47 @@ func testHelper(t *testing.T, data *TestData, podAIPs, podBIPs, podCIPs, podDIPs
 			checkRecordsForFlows(t, data, podAIPs.ipv4.String(), svcC.Spec.ClusterIP, isServiceIPv6, false, true, false, false, checkBandwidth)
 		}
 	})
+
+	// RestartFlowAggregator tests the case where the Flow Aggregator stops: the Flow Exporter should
+	// be able to detect and stop sending records until the Flow Aggregator restarts.
+	t.Run("RestartFlowAggregator", func(t *testing.T) {
+		if err := deleteFlowAggregatorPod(t, testData); err != nil {
+			t.Error(err)
+		}
+		// Run a ToExternalFlowsTest to check whether records can be collected correctly after
+		// restarting the Flow Aggregator Pod
+		runToExternalFlowsTest(t, data, isIPv6, true)
+	})
+}
+
+func runToExternalFlowsTest(t *testing.T, data *TestData, isIPv6 bool, isRestart bool) {
+	// Creating an agnhost server as a host network Pod
+	serverPodPort := int32(80)
+	_, serverIPs, cleanupFunc := createAndWaitForPod(t, data, func(name string, ns string, nodeName string) error {
+		return data.createServerPod(name, testNamespace, "", serverPodPort, false, true)
+	}, "test-server-", "", testNamespace)
+	defer cleanupFunc()
+
+	clientName, clientIPs, cleanupFunc := createAndWaitForPod(t, data, data.createBusyboxPodOnNode, "test-client-", nodeName(0), testNamespace)
+	defer cleanupFunc()
+
+	if !isIPv6 {
+		if clientIPs.ipv4 != nil && serverIPs.ipv4 != nil {
+			if !isRestart {
+				checkRecordsForToExternalFlows(t, data, nodeName(0), clientName, clientIPs.ipv4.String(), serverIPs.ipv4.String(), serverPodPort, isIPv6)
+			} else {
+				checkRecordsForFlowsAfterRestart(t, data, nodeName(0), clientName, clientIPs.ipv4.String(), serverIPs.ipv4.String(), serverPodPort, isIPv6)
+			}
+		}
+	} else {
+		if clientIPs.ipv6 != nil && serverIPs.ipv6 != nil {
+			if !isRestart {
+				checkRecordsForToExternalFlows(t, data, nodeName(0), clientName, clientIPs.ipv6.String(), serverIPs.ipv6.String(), serverPodPort, isIPv6)
+			} else {
+				checkRecordsForFlowsAfterRestart(t, data, nodeName(0), clientName, clientIPs.ipv6.String(), serverIPs.ipv6.String(), serverPodPort, isIPv6)
+			}
```

Comment:
if you pass `isRestart` as a flag, then I feel it's no longer worth trying to unify the test cases with a common `runToExternalFlowsTest` function. I think it just makes the tests harder to read & understand.

Reply:
Makes sense. Updated.